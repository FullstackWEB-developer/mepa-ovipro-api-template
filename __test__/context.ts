import * as cdk from '@aws-cdk/core';
import { readFileSync } from 'fs';
const cdkJson = JSON.parse(readFileSync('./cdk.json', 'utf-8'));

class TestContextProvider extends cdk.Construct {
    constructor(scope: cdk.Construct, id: string) {
        super(scope, id);
        this.node.setContext('account', 'mock');
        this.node.setContext('environment', 'mock1');
        this.node.setContext('defaultRegion', cdkJson.context.defaultRegion);
        this.node.setContext('project', cdkJson.context.project);
        this.node.setContext('accounts', cdkJson.context.accounts);
    }
}

/**Creates a CDK construct with all the required context fro running tests and assignes it into a CDK app. */
export function createCdkTestContext(): cdk.Construct {
    const app = new cdk.App();
    const testContextProvider = new TestContextProvider(app, 'TestContextProvider');
    return testContextProvider;
}
