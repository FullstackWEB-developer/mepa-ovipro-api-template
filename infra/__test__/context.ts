import { Project } from '@alma-cdk/project';
import { Construct } from 'constructs';
import projectProps from '../bin/config';

class TestContextProvider extends Construct {
    constructor(scope: Construct, id: string) {
        super(scope, id);
        this.node.setContext('account', 'mock');
        this.node.setContext('environment', 'mock1');
    }
}

/**Creates a CDK construct with all the required context fro running tests and assignes it into a CDK app. */
export function createCdkTestContext(): Construct {
    const app = new Project(projectProps);
    const testContextProvider = new TestContextProvider(app, 'TestContextProvider');
    return testContextProvider;
}
