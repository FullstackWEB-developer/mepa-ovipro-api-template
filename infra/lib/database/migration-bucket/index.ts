import * as cdk from '@aws-cdk/core';
import * as s3 from '@aws-cdk/aws-s3';
import { Name } from '@almamedia/cdk-tag-and-name';
import { Ac, Ec } from '@almamedia/cdk-accounts-and-environments';
import { pascalCase } from 'change-case';
import { RemovalPolicy } from '@aws-cdk/core';

export class MigrationBucketStack extends cdk.Stack {
    public readonly s3Bucket: s3.Bucket;

    constructor(scope: cdk.Construct, id: string, props: cdk.StackProps) {
        super(scope, id, props);

        const name = `ovipro${Ac.getType(this)}${pascalCase(Ec.getName(this))}migrations`;

        const s3Bucket = new s3.Bucket(this, Name.withProject(this, name), {
            bucketName: name.toLowerCase(),
            blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
            encryption: s3.BucketEncryption.S3_MANAGED,
            autoDeleteObjects: !Ec.isStable(this),
            removalPolicy: Ec.isStable(this) ? RemovalPolicy.RETAIN : RemovalPolicy.DESTROY,
        });

        this.s3Bucket = s3Bucket;
    }
}
