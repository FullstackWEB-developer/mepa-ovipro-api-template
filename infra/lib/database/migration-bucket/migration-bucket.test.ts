import * as cdk from '@aws-cdk/core';
import * as s3 from '@aws-cdk/aws-s3';
import { Sc } from '@almamedia/cdk-accounts-and-environments';
import { SynthUtils } from '@aws-cdk/assert';
import { Name } from '@almamedia/cdk-tag-and-name';
import { createCdkTestContext } from '../../../__test__/context';
import { MigrationBucketStack } from '../migration-bucket';

test('MigrationBucket', () => {
    // GIVEN
    const scope = createCdkTestContext();

    const stack = new MigrationBucketStack(scope, 'MigrationBucketStack', {
        stackName: Name.stack(scope, 'MigrationBucketStack'),
        ...Sc.defineProps(scope, 'Migration bucket for OviPRO infrastructure'),
    });

    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test('MigrationBucket:deletionPolicy', () => {
    // GIVEN
    const scope = createCdkTestContext();

    const { s3Bucket } = new MigrationBucketStack(scope, 'MigrationBucketStack', {
        stackName: Name.stack(scope, 'MigrationBucketStack'),
        ...Sc.defineProps(scope, 'Migration bucket for OviPRO infrastructure'),
    });

    // THEN
    const bucket = s3Bucket.node.defaultChild as s3.CfnBucket;
    expect(bucket.cfnOptions.deletionPolicy).toBe(cdk.CfnDeletionPolicy.DELETE);
});
