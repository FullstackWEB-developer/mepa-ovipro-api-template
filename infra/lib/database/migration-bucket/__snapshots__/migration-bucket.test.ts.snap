// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MigrationBucket 1`] = `
Object {
  "Description": "Mock1 Environment: Migration bucket for OviPRO infrastructure",
  "Parameters": Object {
    "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392ArtifactHashE56CD69A": Object {
      "Description": "Artifact hash for asset \\"4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392\\"",
      "Type": "String",
    },
    "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3BucketBF7A7F3F": Object {
      "Description": "S3 bucket for asset \\"4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392\\"",
      "Type": "String",
    },
    "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3VersionKeyFAF93626": Object {
      "Description": "S3 key for asset version \\"4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": Object {
      "DependsOn": Array [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3BucketBF7A7F3F",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3VersionKeyFAF93626",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4cd61014b71160e8c66fe167e43710d5ba068b80b134e9bd84508cf9238b2392S3VersionKeyFAF93626",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Lambda function for auto-deleting objects in ",
              Object {
                "Ref": "OviproMock1OvipromockMock1migrationsD48EF732",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "OviproMock1OvipromockMock1migrationsAutoDeleteObjectsCustomResourceE9D4C619": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "OviproMock1OvipromockMock1migrationsPolicy7DAC42B3",
      ],
      "Properties": Object {
        "BucketName": Object {
          "Ref": "OviproMock1OvipromockMock1migrationsD48EF732",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "OviproMock1OvipromockMock1migrationsD48EF732": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "OviproMock1OvipromockMock1migrationsPolicy7DAC42B3": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "OviproMock1OvipromockMock1migrationsD48EF732",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "OviproMock1OvipromockMock1migrationsD48EF732",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "OviproMock1OvipromockMock1migrationsD48EF732",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
